lib/ssh_scan/rule_engine.rb

```diff
@@ -0,0 +1,94 @@
+module SSHScan
+  class RuleEngine
+    # Simple one rule matching.
+    def self.matchSingleRule(key, value, result)
+      answer = true
```

Comment:
I agree. I was looking from an AND perspective, but it is certainly possible to do this. Will change.